be/src/runtime/query_context.h

```diff
@@ -109,6 +109,9 @@ class QueryContext {
             _is_cancelled = is_cancelled;
             _ready_to_execute = true;
         }
+        if (query_mem_tracker && is_cancelled) {
+            query_mem_tracker->set_is_query_cancelled(is_cancelled);
```

Comment:
Why here, not in `PlanFragmentExecutor::cancel` and `PipelineFragmentContext::cancel`

seems to break the meaning of `set_ready_to_execute`